### ISSUE TITLE
Jitsi widget doesn't release camera after removing the widget

### ISSUE THREAD

**Issue by AmandineLP** [vector-im/element-web#15703]

**Comment by turt2live** [comment]
Browser information? I can't reproduce this.

**Comment by turt2live** [comment]
reported as nightly

**Comment by turt2live** [comment]
this is reportedly from the hangup button near the composer.

a possible cause is the hangup action racing with the unpin, causing the bug where unpinning causes the camera to lock on with no jitsi widget in sight.

**Comment by thielmal** [comment]
Here's a reproduction description from an internal issue tracker:

At the moment, microphone and camera continue to run unnoticed if the conference is ended through the functions “remove for everyone” in the widget or “end conference” next to the chat field.

**Steps to Reproduce:**

1. Create videoconference widget (button “video call”)
2. “Join Conference”
3. Activate camera in the conference → physical camera LED is active and browser indicates usage of the camera
4. (optional): For observational purposes: Open invite link to the jitsi conference in a new window.
5. Remove widget for everyone or end the conference (specifically through the Element hangup button to the right of the message input, not the Jitsi hangup button)

**Expected Behavior:**
Conference has ended, no further data is transferred through jitsi, camera LED and browser indicator are off.

**Observed Behavior:**
Conference is no longer visible, camera LED and browser indicators continue to be on. In the window opened in step 4 the participant who has “left” is still visible, so data is still being transferred.

**Further Remarks:**

If the end call button inside the widget is used the participanet leaves the conference as expected (camera indicators are off etc.). In Chromium-based browsers (Chrome, Edge) there is still a camera icon in the address bar, but this seems to be normal behavior and only indicates that camera permissions have been granted.

The erroneous behavior was also observed on element.io and in the Docker image “vectorim/element-web:latest” (date: 16. Feb 2021).

**Workaround**
Reload the page.

**Comment by gsouquet** [comment]
It seems that @turt2live removed those buttons from the composer bar as part of matrix-org/matrix-react-sdk#5693

This has the fortunate result of resolving this issue at the same time.

I am going to close this issue and the associated rageshake

**Comment by turt2live** [comment]
I don't believe this is fixed though, @gsouquet  - there are complaints on duplicate issues like https://github.com/vector-im/element-web/issues/16725 about recent versions not giving up the camera. The hangout button is unlikely to be issue (since it isn't there), but there's clearly still *something* going wrong.

Not sure how you'd like to track this.

**Comment by SimonBrandner** [comment]
I think it makes sense to keep #16725 and close this

**Comment by gsouquet** [comment]
@turt2live makes sense, I overlooked the other issue.
Will investigate the root cause of #16725 and keep this one closed as the button in the composer bar do not exist anymore.

**Comment by JonasKress** [comment]
Still seems to be an issue:
https://user-images.githubusercontent.com/13198391/121132304-43b7be00-c831-11eb-8d8f-685ce381bbcc.mp4

**Comment by dbkr** [comment]
Confirmed this is reproducible
